src/WordPress/Hooks.php

```diff
@@ -146,28 +146,36 @@ public function purgeCacheByRelevantURLs($postId)
             }
             $wpDomain = $wpDomainList[0];
 
-            // Do not purge for autosaves or updates to post revisions.
-            if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
-                return;
-            }
+            $postIds = (array) $postIds;
+            $urls = [];
+            foreach ($postIds as $postId) {
+                // Do not purge for autosaves or updates to post revisions.
+                if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
+                    continue;
+                }
 
-            $postType = get_post_type_object(get_post_type($postId));
-            if (!is_post_type_viewable($postType)) {
-                return;
-            }
+                $postType = get_post_type_object(get_post_type($postId));
+                if (!is_post_type_viewable($postType)) {
+                    continue;
+                }
 
-            $savedPost = get_post($postId);
-            if (!is_a($savedPost, 'WP_Post')) {
-                return;
-            }
+                $savedPost = get_post($postId);
+                if (!is_a($savedPost, 'WP_Post')) {
+                    continue;
+                }
 
-            $urls = $this->getPostRelatedLinks($postId);
-            $urls = apply_filters('cloudflare_purge_by_url', $urls, $postId);
+                $urls = array_values(array_unique(array_merge($urls, $this->getPostRelatedLinks($postId))));
+                $urls = apply_filters('cloudflare_purge_by_url', $urls, $postId);
+            }
+            $urls = apply_filters('cloudflare_purge_by_urls', $urls, $postIds);
 
-            if (empty($urls)) {
+            if (!is_array($urls) || empty($urls)) {
```

Comment:
given you've instantiated `$urls = []` at the start of this function, how will the `is_array` check ever fail?

Reply:
See above, user may return anything from the filter

Reply:
an example of what a user may do:

add_filter('cloudflare_purge_by_urls', function($urls) {
   return 'anything_thats_not_an_array';
});